### PR TITLE
Add documentation about the Java binding 

### PR DESCRIPTION
**Pull request by samuel-rufi**
Fixes https://github.com/iotaledger/iota.rs/issues/1098

### CODE REVIEW

File: bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/CreateBlock.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import com.google.gson.JsonObject;
+import org.iota.Client;
+import org.iota.types.Block;
+import org.iota.types.ClientConfig;
+import org.iota.types.ClientException;
+import org.iota.types.ids.BlockId;
+
+public class CreateBlock {
+
+    private static final String DEFAULT_TESTNET_NODE_URL = "https://api.alphanet.iotaledger.net";
```
Comment (Thoralf-M) [inline bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/CreateBlock.java]:
```suggestion
    private static final String DEFAULT_TESTNET_NODE_URL = "https://api.testnet.shimmer.network";
```

File: documentation/docs/libraries/java/how_to/get_block.mdx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import CodeBlock from '@theme/CodeBlock';
+import GetBlock from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/GetBlock.java';
+import GetBlockRaw from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/GetBlockRaw.java';
+
+<CodeBlock className="language-java">
+  {GetBlock}
+</CodeBlock>
+
+<CodeBlock className="language-java">
+    {GetBlockRaw}
```
Comment (Thoralf-M) [inline documentation/docs/libraries/java/how_to/get_block.mdx]: Align indentation

File: documentation/sidebars.js [hunk 3]
```diff
@@ -148,8 +148,48 @@ module.exports = {
...
+        },
+        {
+          type: 'doc',
+          id: 'libraries/java/how_to/get_node_info',
+          label: 'Get Node Info',
+        },
+        {
+          type: 'doc',
+          id: 'libraries/java/how_to/get_output',
+          label: 'Get A Output',
```
Comment (Thoralf-M) [inline documentation/sidebars.js]:
```suggestion
          label: 'Get An Output',
```

File: documentation/docs/libraries/java/how_to/get_block.mdx (hunk 2 above)
Comment (samuel-rufi) [reply documentation/docs/libraries/java/how_to/get_block.mdx]: I did format all the doc files with the IDE formatter, is this fine now for you or should I format it differently?

Comment (Thoralf-M) [reply documentation/docs/libraries/java/how_to/get_block.mdx]: Should be fine

File: documentation/docs/libraries/java/how_to/create_block.mdx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+---
+title: How To Create A Block
+---
+
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/create_block.mdx:4]: For this and all other, just put one blank line

File: documentation/sidebars.js [hunk 5]
```diff
@@ -148,8 +148,48 @@ module.exports = {
         },
         {
           type: 'doc',
-          id: 'libraries/java/examples',
-          label: 'Examples'
+          id: 'libraries/java/how_to/build_output',
```
Comment (thibault-martinez) [inline documentation/sidebars.js]: Do the same order as the rust version

File: documentation/sidebars.js [hunk 6]
```diff
@@ -148,8 +148,48 @@ module.exports = {
         },
         {
           type: 'doc',
-          id: 'libraries/java/examples',
```
Comment (thibault-martinez) [inline documentation/sidebars.js:151]: You need to have an actual how to category, look at the rust version

File: documentation/docs/libraries/java/how_to/create_block.mdx (hunk 4 above)
Comment (samuel-rufi) [reply documentation/docs/libraries/java/how_to/create_block.mdx:4]: fixed

File: documentation/sidebars.js (hunk 6 above)
Comment (samuel-rufi) [reply documentation/sidebars.js:151]: fixed

File: documentation/sidebars.js (hunk 5 above)
Comment (samuel-rufi) [reply documentation/sidebars.js]: fixed

**Review by Thoralf-M — DISMISSED** [review]

File: documentation/docs/libraries/java/getting_started.md [hunk 7]
```diff
@@ -2,168 +2,53 @@
...
 
-## Maven
-We need to make sure the file is added to `java.library.path` before building or running.
-To do this, we add/update the section below to our `pom.xml`
+1) Clone the repository: `git clone https://github.com/iotaledger/iota.rs`
+2) Change directory: `cd iota.rs/bindings/java/iota-client-java`
+3) If needed make `gradlew` executable: `chmod +x gradlew`
+4) Build your JAR: `./gradlew jar`
+5) Find the produced JAR in: `build/libs/`
+6) Add the JAR as a library to your Java project.
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/getting_started.md]: Do a proper markdown list, this doesn't look good on the doc

Comment (samuel-rufi) [reply documentation/docs/libraries/java/getting_started.md]: fixed

File: documentation/docs/libraries/java/how_to/build_output.mdx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import BuildAliasOutput
+    from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/output_builder/BuildAliasOutput.java';
+import BuildBasicOutput
+    from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/output_builder/BuildBasicOutput.java';
+import BuildFoundryOutput
+    from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/output_builder/BuildFoundryOutput.java';
+import BuildNftOutput
+    from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/output_builder/BuildNftOutput.java';
+
+<CodeBlock className="language-java">
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/build_output.mdx:16]: Put a sentence before each like "How To Build A Basic Output" etc

File: documentation/docs/libraries/java/how_to/generate_addresses.mdx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+---
+title: How To Generate Addresses
+---
+
+import CodeBlock from '@theme/CodeBlock';
+
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/generate_addresses.mdx]: For this one and all, no blank line between imports

File: bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/CreateBlock.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.google.gson.JsonObject;
+import org.iota.Client;
+import org.iota.types.Block;
+import org.iota.types.ClientConfig;
+import org.iota.types.ClientException;
+import org.iota.types.ids.BlockId;
+
+public class CreateBlock {
+
```
Comment (thibault-martinez) [inline bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/CreateBlock.java]: Your use of blank lines in the examples is very inconsistent, would be nice if you could harmonise a bit

File: documentation/docs/libraries/java/how_to/generate_addresses.mdx (hunk 9 above)
Comment (samuel-rufi) [reply documentation/docs/libraries/java/how_to/generate_addresses.mdx]: fixed

File: documentation/docs/libraries/java/how_to/build_output.mdx (hunk 8 above)
Comment (samuel-rufi) [reply documentation/docs/libraries/java/how_to/build_output.mdx:16]: fixed

File: documentation/docs/libraries/java/how_to/get_output.mdx [hunk 11]
```diff
@@ -0,0 +1,11 @@
+---
+title: How To Get A Output
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/get_output.mdx]:
```suggestion
title: How To Get An Output
```

File: documentation/docs/libraries/java/how_to/post_block.mdx [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+title: How To Post A Block
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import PostBlock
+    from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PostBlock.java';
+import PostBlockRaw
+    from '!!raw-loader!./../../../../../bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PostBlockRaw.java';
+
+## How To Post A Block as JSON
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/post_block.mdx]:
```suggestion
## How To Post A Block As JSON
```

File: documentation/docs/libraries/java/how_to/post_block.mdx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+<CodeBlock className="language-java">
+    {PostBlock}
+</CodeBlock>
+
+## How To Post A Block as Raw bytes
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/post_block.mdx]:
```suggestion
## How To Post A Block As Raw bytes
```

File: bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/CreateBlock.java (hunk 10 above)
Comment (samuel-rufi) [reply bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/CreateBlock.java]: fixed

File: bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/GenerateAddresses.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+        // Generate the addresses.
+        MnemonicSecretManager secretManager = new MnemonicSecretManager("endorse answer radar about source reunion marriage tag sausage weekend frost daring base attack because joke dream slender leisure group reason prepare broken river");
+        String[] addresses = client.generateAddresses(secretManager, new GenerateAddressesOptions().withRange(0, 5));
+
+        // Print the addresses.
+        for (String address : addresses) {
+            System.out.println(address);
+        }
+
```
Comment (thibault-martinez) [inline bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/GenerateAddresses.java]:
```suggestion
```

File: documentation/docs/libraries/java/how_to/build_output.mdx [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    {BuildAliasOutput}
+</CodeBlock>
+
+## How To Build An Basic Output
+<CodeBlock className="language-java">
+    {BuildBasicOutput}
+</CodeBlock>
+
+## How To Build An Foundry Output
```
Comment (thibault-martinez) [inline documentation/docs/libraries/java/how_to/build_output.mdx]:
```suggestion
## How To Build A Foundry Output
```

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/java/iota-client-java/README.md [hunk 16]
```diff
@@ -2,21 +2,49 @@
...
+        System.out.println(response.getNodeInfo());
+    }
+}
+```
+
+# Documentation
+
+Please visit the [examples](../../../documentation/docs/libraries/java/getting_started.md) page for more information on using the IOTA Java Client Library.
+More examples on how to use the library can be found [here](examples/ExampleProject/src). In addition, since the IOTA Java library is similar to the IOTA Rust library, you might also want to look
+into Rust examples.
```
Comment (Thoralf-M) [inline bindings/java/iota-client-java/README.md]: The newline here looks a little bit weird, maybe put the whole sentence in a new line?

Comment (samuel-rufi) [reply bindings/java/iota-client-java/README.md]: fixed

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**